src/methods/wavelet_based.jl

```diff
@@ -0,0 +1,133 @@
+export WLS
+using Wavelets
+
+"""
+    WLS(wt::Wavelets.WT.OrthoWaveletClass, surromethod::Surrogate)
+
+A wavelet surrogate generated by taking the maximal overlap discrete 
+wavelet transform (MODWT) of the signal, shuffling detail 
+coefficients at each dyadic scale using the provided `surromethod`,
+then taking the inverse transform to obtain a surrogate.
+
+Based on Keylock (2006)[^Keylock2006], but in contrast to the original 
+implementation, you may choose to use any surrogate method from this 
+package to perform the randomization of the detail coefficients at each 
+dyadic scale. The iterative steps after rank ordering (step [v] in [^Keylock2006])
+are not performed in this implementation.
+
+If `surromethod == IAAFT()`, the wavelet surrogates preserves the mean 
+and variance structure of the signal, but randomises nonlinear 
+properties of the signal (i.e. Hurst exponents)[^Keylock2006]. In contrast 
+to IAAFT surrogates, the IAAFT shuffled wavelet surrogates also 
+preserves nonstationarity. 
+
+To deal with nonstationary signals, Keylock (2006) recommends using a 
+wavelet with a high number of vanishing moments. Thus, default here is 
+a Daubechies wavelet with 16 vanishing moments.
+
+[^Keylock2006]: C.J. Keylock (2006). "Constrained surrogate time series with preservation of the mean and variance structure". Phys. Rev. E. 73: 036707. doi:10.1103/PhysRevE.73.036707.
+"""
+struct WLS{WT <: Wavelets.WT.OrthoWaveletClass} <: Surrogate
+    wt::WT
+    surromethod::Surrogate # should preserve values of the original series
+end
+
+# Keylock (2006) recommends a wavelet with high number of vanishing moments,
+# so use Daubechies wavelets with 16 vanishing moments as default.
+WLS() = WLS(Wavelets.WT.Daubechies{16}(), AAFT())
+WLS(surromethod::Surrogate) = WLS(Wavelets.WT.Daubechies{16}(), surromethod)
```

Comment:
I am not sure this order of arguments is the best... If you just define a single 
```
WLS(surr = AAFT(), wave = Daubechies{16}())
```
you can use the same method in the docstring, thus showing the defaults as well. I was bit confused reading the documentation page and seeing a `<:Surrogate` passed as the first argument everywhere.